keras/utils/dataset_utils.py

```diff
@@ -14,14 +14,137 @@
 # ==============================================================================
 """Keras image dataset loading utilities."""
 
+
 import tensorflow.compat.v2 as tf
 # pylint: disable=g-classes-have-attributes
 
 import multiprocessing
 import os
+import time
 
 import numpy as np
+from tensorflow.python.util.tf_export import keras_export
+
+@keras_export('keras.utils.split_dataset')
+def split_dataset(dataset, 
+                  left_size=None, 
+                  right_size=None, 
+                  shuffle=False, 
+                  seed=None):
+  """Split a dataset into a left half and a right half (e.g. training / validation).
+  
+  Args:
+      dataset: A `tf.data.Dataset` object or 
+               a list/tuple of arrays with the same length.   
+      left_size: If float, it should be in range `[0, 1]` range 
+                 and signifies the fraction of the data to pack in 
+                 the left dataset. If integer, it signifies the number 
+                 of samples to pack in the left dataset. 
+                 If `None`, it defaults to the complement to `right_size`.       
+      right_size: If float, it should be in range `[0, 1]` range 
+                  and signifies the fraction of the data to pack 
+                  in the right dataset. If integer, it signifies 
+                  the number of samples to pack in the right dataset.
+                  If `None`, it defaults to the complement to `left_size`.     
+      shuffle: Boolean, whether to shuffle the data before splitting it.
+      seed: A random seed for shuffling.
+
+  Returns:
+      A tuple of two `tf.data.Dataset` objects: the left and right splits.
+  """
+  
+  if dataset and not isinstance(dataset,(tf.data.Dataset,list)):
+    raise TypeError('`dataset` must be either a tf.data.Dataset object'
+                     'or a list/tuple of arrays',f'Got {type(dataset)}')
+    
+  dataset_as_list = []
+  data_size_warning_flag = False
+  start_time = time.time()
+  
+  for datum in list(dataset):
```

Comment:
I'm noticing that we should handle the case where the dataset is a `tf.data.Dataset` that is already batched. In such cases we should iterate over the batches, split the dataset at the batch level, then merge the batches into new datasets by concatenating on the 0th axis. That way you get two batched datasets with the same batch size as the original one.

It is not possible (I think?) to tell in advance whether a dataset is already batched or not, so we would have to add an argument for it I think, e.g. `batched=False`.